bin/tronctl

```diff
@@ -147,12 +179,50 @@ def control_objects(args):
             )
             raise SystemExit(f"Error: {e}{suggestions}")
 
-        data = dict(command=args.command)
-        if args.command == "start" and args.run_date:
-            data['run_time'] = str(args.run_date)
-        if args.command == "retry":
-            data['use_latest_command'] = int(args.use_latest_command)
-        yield request(urljoin(args.server, tron_id.url), data)
+        if args.command == "skip-and-publish":
+            # this command is more of a pseudo-command - skip and publish are handled in two different resources
+            # and changing the API would be painful, so instead we call skip + publish separately from the client
+            # (i.e., this file) to implement this functionality
+            if request(
+                url=urljoin(args.server, tron_id.url),
+                data={"command": "skip"}
+            ):
+                # a single action can have 0..N triggers to publish and these can be arbitrarily named, so we need to
+                # query the API and figure out what triggers exist
+                triggers = _get_triggers_for_action(server=args.server, action_identifier=identifier)
+                if triggers is None:
+                    # tronctl allows for multiple ids to be passed in for commands, but we want to make it obvious that
+                    # we failed to get the downstream triggers so that users know to take action (and don't miss the
+                    # error amongst other tronctl output).
+                    print("Encountered error getting triggers to publish.")
+                    yield False
+                elif not triggers:
+                    # TODO: should we check this up-front and refuse to skip if there are no triggers that will be
+                    # published rather than carry on under the assumption that the user copy-pasted/typo'd the identifier?
+                    print(f"{identifier} has no triggers to publish.")
+                    yield True
+
+                # TODO: this loop should use event_publish(), but we'd need to refactor how the CLI works and stop passing
```

Comment:
If we stopped passing around the full set of args, we could also make this a separate command function rather than adding this to `control_objects`, right?

I agree that would make the code a lot nicer, but I'm ok with following up with it later.

Reply:
yup!